packages/web3-core/src/web3_request_manager.ts

```diff
@@ -169,28 +171,66 @@ export class Web3RequestManager<
 				response = error as JsonRpcResponse<ResponseType>;
 			}
 
-			return this._processJsonRpcResponse(payload, response);
+			return this._processJsonRpcResponse(payload, response, { legacy: false, error: false });
+		}
+
+		if (isEIP1193Provider(provider)) {
```

Comment:
I think we need later to consider refactoring to put the difference between every provider inside the provider class. So, we would not have to handle the differences in if statements.

Reply:
Yes a lot can be refactored but most of that will be deprecated soon anyway.